src/App.tsx

```diff
@@ -9,22 +9,46 @@ import { Page } from './pages.ts';
 
 import './App.css';
 
-const pageComponents = {
-    [Page.ABOUT]: <About />,
-    [Page.PRODUCTS]: <Products />,
-} as const;
-
 function App() {
     const [currentPage, setCurrentPage] = useState<Page>(Page.PRODUCTS);
+    const [isDarkTheme, setIsDarkTheme] = useState(() => {
+        const savedTheme = localStorage.getItem('theme');
+        return savedTheme ? JSON.parse(savedTheme) : window.matchMedia('(prefers-color-scheme: dark)').matches;
+    });
+
+    useEffect(() => {
+        const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+        const handleChange = () => {
+            setIsDarkTheme(mediaQuery.matches);
+        };
+        mediaQuery.addEventListener('change', handleChange);
+        return () => {
+            mediaQuery.removeEventListener('change', handleChange);
+        };
+    }, []);
+
+    useEffect(() => {
+        document.body.className = isDarkTheme ? 'dark-theme' : 'light-theme';
+        localStorage.setItem('theme', JSON.stringify(isDarkTheme));
+    }, [isDarkTheme]);
```

Comment:
good, buit suggest u to create a custom hook for this logic
example: 
```ts
const { isDarkTheme, handleThemeChange, ...anyDataRelatedToThemeChange} = useTheme();
```

the idea is to create a hook which will handle all theme switch logic

Reply:
![Screenshot_1](https://github.com/Allocene/fe-react-2024/assets/120042417/36ab919c-4470-46fd-b94b-65b20b0e0243)
Without an empty first one, I can't stylize the dropdown list(